kubernetes/secret.go

```diff
@@ -39,12 +41,34 @@ type RemoteSecret struct {
 }
 
 func GetRemoteSecret(path string) (*RemoteSecret, error) {
-	secret := &RemoteSecret{}
 	secretFile, err := ioutil.ReadFile(path)
 	if err != nil {
 		return nil, err
 	}
-	err = yaml.Unmarshal(secretFile, &secret)
+	return ParseRemoteSecretBytes(secretFile)
+}
+
+// GetSecrets returns a list of secrets for a given namespace.
+// If selectorLabels is defined, the list will only contain services matching
+// the specified label selector.
+func (in *K8SClient) GetSecrets(namespace string, labelSelector string) ([]core_v1.Secret, error) {
```

Comment:
For me, it seems more natural to have this methods into the `client.go` file because there are all the methods dealing with pure k8s objects. This secret files sounds a bit more a helpers around dealing with secrets.

Reply:
I think I'm not sure to understand the structure that good :) because `client.go` for me, looks more like interface definitions and a few helpers related to creating the client. So, I this code would be like an "outlier" if moved to `client.go` :/

Perhaps, it's something else?